### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by hoc081098**
* Migrate this package to null safety.
* Update SDK constraints: `>=2.12.0-0 <3.0.0` based on beta release guidelines.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.2.0-nullsafety.0
+  * Migrate this package to null safety.
+  * Sdk constraints: `>=2.12.0-0 <3.0.0` based on beta release guidelines.
+
```
Comment (zeshuaro) [inline CHANGELOG.md:4]: Please remove changes in this file, I'll update it when I publish the package

File: analysis_options.yml [hunk 2]
```diff
@@ -1 +1 @@
-include: package:pedantic/analysis_options.yaml
+include: package:pedantic/analysis_options.1.9.0.yaml
```
Comment (zeshuaro) [inline analysis_options.yml:1]: Should this version be `1.10.0` for null safety?

File: example/ios/Podfile.lock [hunk 3]
```diff
@@ -14,9 +14,9 @@ EXTERNAL SOURCES:
...
 
 SPEC CHECKSUMS:
-  Flutter: 0e3d915762c693b495b44d77113d4970485de6ec
+  Flutter: 434fef37c0980e73bb6479ef766c45957d4b510c
   package_info: 873975fc26034f0b863a300ad47e7f1ac6c7ec62
 
 PODFILE CHECKSUM: a75497545d4391e2d394c3668e20cfb1c2bbd4aa
 
-COCOAPODS: 1.10.0
+COCOAPODS: 1.9.3
```
Comment (zeshuaro) [inline example/ios/Podfile.lock:22]: I'll prefer to keep things up-to-date, but if you don't want to update your CocoaPods, that's fine too I'll fix it up afterwards

File: example/pubspec.yaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
 name: google_api_headers_example
 description: Demonstrates how to use the google_api_headers plugin.
-version: 1.0.0+1
+version: 2.0.0+2
```
Comment (zeshuaro) [inline example/pubspec.yaml:3]: Don't worry about changing the version of the example app

File: pubspec.yaml [hunk 5]
```diff
@@ -1,22 +1,23 @@
 name: google_api_headers
 description: A Flutter plugin for getting the headers required for calling Google APIs with an app restricted key.
-version: 0.1.1+2
+version: 0.2.0-nullsafety.0
```
Comment (zeshuaro) [inline pubspec.yaml:3]: Don't worry about updating the version, I'll update it before I publish the package

File: analysis_options.yml (hunk 2 above)
Comment (hoc081098) [reply analysis_options.yml:1]: File `analysis_options.1.10.0.yaml` does not exists, https://github.com/google/pedantic/tree/master/lib

File: test/google_api_headers_test.mocks.dart [hunk 6]
```diff
@@ -0,0 +1,27 @@
+// Mocks generated by Mockito 5.0.0-nullsafety.7 from annotations
```
Comment (zeshuaro) [inline test/google_api_headers_test.mocks.dart:1]: I prefer to not check in auto-generated files into git, can you do the followings: 
- [ ] Update `.gitignore` to ignore `*.mocks.dart` files
- [ ] Update `.github/workflows/analyze_test.yml` to include a step to generate this file

If you're not sure about anything, let me know and I can help you out

File: example/lib/main.dart [hunk 7]
```diff
@@ -12,7 +13,7 @@ class MyApp extends StatefulWidget {
 }
 
 class _MyAppState extends State<MyApp> {
-  Map<String, dynamic> _headers;
+  Map<String, String>? _headers;
```
Comment (zeshuaro) [inline example/lib/main.dart:16]: Is changing `dynamic` to `String` required as part of the null safety migration?

**Comment by zeshuaro** [comment]
@hoc081098 Thank you for putting the effort to migrate this package for null safety, please check out my comments 🙂

File: test/google_api_headers_test.mocks.dart (hunk 6 above)
Comment (hoc081098) [reply test/google_api_headers_test.mocks.dart:1]: I wil ignore it

**Comment by zeshuaro** [comment]
@hoc081098 I've just updated the CI config, if you merge `main` branch into your branch and push again, it should trigger the checks again hopefully

File: test/google_api_headers_test.dart [hunk 8]
```diff
@@ -18,8 +20,8 @@ void main() {
     'google_api_headers',
   );
 
-  final platform = MockPlatform();
-  List<MethodCall> log;
+  final platform = MockMyPlatform();
+  late List<MethodCall> log;
```
Comment (zeshuaro) [inline test/google_api_headers_test.dart:24]: Can this be `late final`?

File: example/lib/main.dart (hunk 7 above)
Comment (hoc081098) [reply example/lib/main.dart:16]: `dynamic` is like `Object?` in nnbd, but our map does not contains null value

Comment (zeshuaro) [reply example/lib/main.dart:16]: I see that makes sense

File: test/google_api_headers_test.dart (hunk 8 above)
Comment (hoc081098) [reply test/google_api_headers_test.dart:24]: Cannot, because `log` is initialized many times.

Comment (zeshuaro) [reply test/google_api_headers_test.dart:24]: You're right, I forgot it's initialized in every test, all good then

**Comment by zeshuaro** [comment]
Everything looks good now, I'm merging it into the `null-safety` branch and publish it later. Thanks again for your contribution!

**Pull request merged**